projects/js-packages/image-guide/src/MeasurableImage.ts

```diff
@@ -11,15 +11,19 @@ export class MeasurableImage {
 	readonly node: HTMLElement | HTMLImageElement;
 	private getURLCallback: SourceCallbackFn;
 
+	public fetch = fetch;
+
 	/**
 	 * Constructor.
 	 *
 	 * @param {HTMLElement | HTMLImageElement} node -  The DOM Element that contains the image.
 	 * @param {SourceCallbackFn} getURL             -  A function that takes in the node and returns the URL of the image.
+	 * @param {(input: string, init?: Array) => Promise<Response>} fetchFn                               -  A function that fetches a URL and returns a Promise.
 	 */
-	constructor( node: HTMLElement | HTMLImageElement, getURL: SourceCallbackFn ) {
+	constructor( node: HTMLElement | HTMLImageElement, getURL: SourceCallbackFn, fetchFn = fetch ) {
```

Comment:
One thing I'm fussy about: I like to specify types for all arguments. As a pro-tip, you can define a type at the top of the file like:

```
type FetchFn = ( input: URL | RequestInfo, init?: RequestInit ) => Promise<Response>;
```

Then whenever you specify the type -- both in the param list here, and in the @param comment above, you can just say `FetchFn`.

Reply:
Thanks, I wasn't sure what type to put in there.